src/Components/Layout.tsx

```diff
@@ -58,12 +63,37 @@ export function Layout() {
     }
   }
 
+  function toMilliseconds() {
+    var milliseconds = 0
+    if (refreshIntervalUnits == 'Seconds') {
+      milliseconds = refreshInterval*1000
+    } else if (refreshIntervalUnits == 'Minutes') {
+      milliseconds = refreshInterval*1000*60
+    } else if (refreshIntervalUnits == 'Hours') {
+      milliseconds = refreshInterval*1000*60*60
+    }
+    if (refreshInterval > 1000*60*60*24) {
+      milliseconds = 1000*60*60*24-1
+    }
+    return milliseconds
+  }
+
+  function timer() {
+    if (timerRunning == true) {
+      return <Timer initialTime={toMilliseconds()} direction="backward">{({timerState}) => (<React.Fragment>{"Refresh in "}<Timer.Hours/>{":"}<Timer.Minutes/>{":"}<Timer.Seconds/></React.Fragment>)}</Timer>
+    } else {
+      return <div>Refesh</div>
+    }
+  }
+
     return (
       <div className='layout'>
         <Tabs onSelect={index => toggleDropdown(index)}>
           <TabList className='header'>
             <Tab>
-              <Button color='primary' variant='contained' className='header-button'>Menu</Button>
+              <Button color='primary' variant='contained' className='header-button' >
+              {timer()}

```

Comment:
Can you reduce the number of functions involved and include the above code directly? Or create an additional component?